### DISCUSSION TITLE
Ender Chest Upgrade

### DISCUSSION THREAD

**Discussion by LoveyDov3** [P3pp3rF1y/SophisticatedBackpacks#1221] [Ideas]
Very simply put, a way to incorporate the Ender Chest into the backpack as an upgrade. I think if it were done in the same way as the fluid upgrade, where you would need 'x' amount of open slots to use the upgrade. That way the open slots could be either colored differently or highlighted in a new border to show that they are being used for the upgrade.

I would think using the number of slots that aren't either used for locked items or highlighted nearer to the bottom would make the most since. So as not to disrupt the Compacting Upgrade. Having it also be like the Crafting Table upgrade as a pop out is also a viable option too.

Another, probably way more complex idea I had for this and as well as for other upgrade slots would be to implement the same functionality as the highlight feature to determine where and what slots will be used. So if you just wanted to use the first 5 slots in your Ender Chest to stay synced with the upgrade slot, you would highlight 5 slot in the backpack and it would sync with the respective slots.

The Ender Chest is the one thing I have wanted to see utilized in some way in your mod ever since I started using them. With or with out the addition, you still have the best, fully fleshed out backpack/storage mod BAR NONE!

Thank you for your time!

V/R

LoveyDov3

**Comment by P3pp3rF1y** [comment]
I have a bit different plan - ender modification / linker or something similar to that. Will be used with backpacks / storage to link the two together, will require the same tier or possibly same number of slots, but when linked the two (or more) backpacks/storages will share inventory. So you will be able to use this as a shared backpack with another player or link backpack to barrel and use the backpack as drop off with the barrel being extracted from at your base and many more things that can be done with this. Hopefully I will have time to get to this some time soon.

**Reply by LoveyDov3** [reply]
Yeah that is a awesome solution to be honest! Is the idea to also have a filter where we would drag and drop items to act as the respective channels? Similar to how the Create mod does this with the 'Redstone Link' where you can have one link be 'diamond & coal' and another could use one slot for just 'Jungle Saplings'? 

I am still on 1.20.1 so I'm not sure if you'll drop this feature on this version but either way I am eager to see what you come up with!

**Reply by P3pp3rF1y** [reply]
So my idea is a new "linking item" that you right click on / craft with storage / backpack that ties the item to that and then you do the same on another storage / backpack which consumes the item and links the two together (or perhaps doesn't even consume allowing you to link more than one using one link item).

**Reply by LoveyDov3** [reply]
I'm eager to see this new upgrade slot in action! 🤗

**Reply by cpebud** [reply]
This sounds like an awesome idea! I was wanting something like this to pump in food items into a backpack and using the auto feeder to have an endless supply of food. Or an auto export to a mass storage feature while mining. Looking forward to seeing this added. 

I was curious if there has been any progress towards this?
